framework/plugins/plugin.go

```diff
@@ -159,6 +160,51 @@ func resolvePlugins(pluginsDir, assetsDir, downloadsDir string, osArch osarch.OS
 	return nil, errors.New(strings.Join(errStringsParts, "\n"+strings.Repeat(" ", pluginsinternal.IndentSpaces)))
 }
 
+func groupByProductAndGroup(plugins []godellauncher.SinglePluginParam) map[string][]godellauncher.SinglePluginParam {
+	pluginByGroupAndProduct := make(map[string][]godellauncher.SinglePluginParam)
+	for _, currPlugin := range plugins {
+		groupKey := currPlugin.LocatorWithChecksums.GroupAndProductString()
+		_, ok := pluginByGroupAndProduct[groupKey]
+		if !ok {
+			pluginByGroupAndProduct[groupKey] = make([]godellauncher.SinglePluginParam, 0)
+		}
+		singlePluginList := pluginByGroupAndProduct[groupKey]
+		pluginByGroupAndProduct[groupKey] = append(singlePluginList, currPlugin)
+	}
+	return pluginByGroupAndProduct
+}
+
+// dedupePlugins will remove any plugins that added by the configuration provider iff there is a plugin
+// present from the pluging portion of configuration that contains the same group and product name
+func dedupePlugins(plugins []godellauncher.SinglePluginParam) []godellauncher.SinglePluginParam {
+	var allPlugins []godellauncher.SinglePluginParam
+	pluginMap := groupByProductAndGroup(plugins)
+	for _, pluginsForProductAndGroup := range pluginMap {
+		noOverride := noOverridePresent(pluginsForProductAndGroup)
+		if noOverride {
```

Comment:
The loop logic is the same over-all, so would push this check down into the loop logic instead:

```go
for _, singlePlugin := range pluginsForProductAndGroup {
	if noOverride || singlePlugin.FromPluginConfig {
}
```